jobs/integr8ly/ocp4/install/image-deploy/Jenkinsfile

```diff
@@ -0,0 +1,175 @@
+Boolean selfSigned = params.SELF_SIGNED_CERTS
+
+String clusterSuffix = isPds ? "cluster-${CLUSTER_NAME}.${CLUSTER_NAME}.${CLUSTER_DOMAIN}" : "${CLUSTER_NAME}.${CLUSTER_DOMAIN}"
+
+String clusterAPI = "https://api.${clusterSuffix}:6443"
+
+String selfSignedCerts = selfSigned ? "true" : "false"
+
+String installationName = "example-installation"
+
+String installationNamespace = "integreatly"
+
+String expectedOcCommandOutput = ""
+
+def err = null
+
+node('cirhos_rhel7') {
+    timeout(75) { ansiColor('gnome-terminal') { timestamps {
+        try {
+            currentBuild.displayName = "${currentBuild.displayName} ${CLUSTER_NAME}"
+
+            stage('Clone integreatly-operator repository') {
+                dir('integreatly-operator') {
+                    checkout([
+                        $class: 'GitSCM',
+                        branches: [[name: INTEGREATLY_OPERATOR_BRANCH]],
+                        doGenerateSubmoduleConfigurations: false,
+                        extensions: [],
+                        userRemoteConfigs: [[url: INTEGREATLY_OPERATOR_REPOSITORY]]
+                    ])
+
+                    sh "git clean -xdf"
+                    sh "oc login ${clusterAPI} --insecure-skip-tls-verify=true -u ${ADMIN_USERNAME} -p ${ADMIN_PASSWORD}"
+                } // dir
+            } // stage
+
+            stage ('Cleanup resources') {
+                dir('integreatly-operator') {
+                    sh(
+                        returnStdout: false,
+                        script: """
+                            oc project integreatly | true
+                            oc delete secret oauth-client-secrets | true
+                            oc delete installation example-installation --timeout=60s --wait | true
+                            oc delete crd applicationmonitorings.applicationmonitoring.integreatly.org | true
+                            oc delete crd blackboxtargets.applicationmonitoring.integreatly.org | true
+                            oc delete crd grafanadashboards.integreatly.org | true
+                            oc delete crd grafanadatasources.integreatly.org | true
+                            oc delete crd grafanas.integreatly.org | true
+                            oc delete crd installations.integreatly.org | true
+                            oc delete crd webapps.integreatly.org | true
+                            oc delete CatalogSourceConfig installed-integreatly-operator -n openshift-marketplace | true
+                            oc delete OperatorSource integreatly-operators -n openshift-marketplace | true
+                            oc delete ClusterRole integreatly-operator | true
+                            oc delete ClusterRoleBinding integreatly-operator | true
+                            oc delete ClusterRoleBinding integreatly-operator-cluster-view | true
+                            oc delete namespace ${installationNamespace} --timeout=240s --wait | true
```

Comment:
Just wondering - is this standard cleanup procedure ??

Reply:
yes. There are more than likely all resources created. There are make targets to do something similar, but they change and can become unreliable. I wouldn't personally mind not having the cleanup stage, but sometimes when doing manual cleanup, some resources are left over, which would then cause the installation to fail.